### PR TITLE
Replaced PUT with PATCH action

### PR DESCRIPTION
**Pull request by hsong-rh**
Based on the legacy definition, PUT is used to update all attributes of the resource, and PATCH is for individuals. In this case Approval API should use PATCH, rather than PUT, to update resources.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -10,7 +10,7 @@
 =end
 Rails.application.routes.draw do
 
-  mount SwaggerUiEngine::Engine, :at => '/open-api'
+  mount SwaggerUiEngine::Engine, at: '/api'
```
Comment (gmcculloug) [inline config/routes.rb]: This seems unrelated to the Put/Patch change and has switched back and forth a few times on this file.  service_portal-api is currently using `open-api` https://github.com/ManageIQ/service_portal-api/blob/master/config/routes.rb#L13

What's the correct value here?

cc @syncrou

Comment (syncrou) [reply config/routes.rb]: I changed it to `open-api` because of the versions pr.  The base path `/api/*` wasn't being recognized correctly by the rails engine, plus I wanted to differentiate between api calls `/adi` and Swagger UI `/open-api`

File: config/routes.rb [hunk 2]
```diff
@@ -10,7 +10,7 @@
 =end
 Rails.application.routes.draw do
 
-  mount SwaggerUiEngine::Engine, :at => '/open-api'
+  mount SwaggerUiEngine::Engine, at: '/open-api'
```
Comment (bzwei) [inline config/routes.rb]: why not rollback to `:at => '/open-api'` so that this diff does not appear at all?

**Comment by bzwei** [comment]
@hsong-rh we may need another PR to clean up all endpoints. Some may not support full CRUD.

File: config/routes.rb (hunk 2 above)
Comment (hsong-rh) [reply config/routes.rb]: Done.

**Comment by hsong-rh** [comment]
@bzwei Yes, I'll create a PR to cleanup some extra endpoints.

**Comment by gmcculloug** [comment]
@hsong-rh @bzwei Please add a more detail into the description as to what promoted this change.  It helps reviewers and future readers to understand the reasoning.  Thanks.

**Comment by hsong-rh** [comment]
@gmcculloug The description is updated with more details.
